### PR TITLE
[EASI-2152] Front End for Ready for Review implementation

### PR DESCRIPTION
**Pull request by garyjzhao**
# EASI-2152

Figma: https://www.figma.com/file/M72DReE73SSXMN48WZfaKN/MINT---New-Model-Process-(Model-Intake)?node-id=2925%3A219994

## Changes and Description

- Adds a section to end of each task list item for user to indicate that the task list is ready for review
- Change 2

## How to test this change

1. Navigate to the end of each task list
2. New model status box will appear on the last page of each model plan section. 
3. Check box label will read, "This section of the Model Plan (Section name) is ready for review."
image

5. Checking this box updates the status on the task list to "Ready for review"
image

7. Navigate to the last page in the same task list item to see that the checkbox is checked and User can see when it was marked for review and who marked it.  This date should always be the date this box is checked even if other changes are made to the form after it's checked.
image

9. Unchecking this box will return the status on the task list to "In progress" but the copy should still be there
image
image

11. Finally, navigate through it one more time to see check status box after it's been unchecked, the line for who originally checked it is gone. 
image

## PR Author Review Checklist

- [ ] Met the ticket's acceptance criteria, or will meet them in a subsequent PR.
- [ ] Added or updated tests for backend resolvers or other functions as needed.
- [ ] Added or updated client tests for new components, parent components, functions, or e2e tests as necessary.
- [ ] Tested user-facing changes with voice-over and the [rotor menu](https://support.apple.com/guide/voiceover/with-the-voiceover-rotor-mchlp2719/mac)

## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.

### CODE REVIEW

File: src/views/ModelPlan/TaskList/index.tsx [hunk 1]
```diff
@@ -113,10 +113,11 @@ const TaskList = () => {
...
   const renderBasicsStatus = (): TaskStatus => {
     if (
-      basics.status === TaskStatus.COMPLETE &&
-      milestones.status === TaskStatus.COMPLETE
+      // * TEMPORARY SOLUTION to render "READY FOR REVIEW" on Task List page, until better solution from BE
+      // basics.status === TaskStatus.READY_FOR_REVIEW &&
+      milestones.status === TaskStatus.READY_FOR_REVIEW
     ) {
-      return TaskStatus.COMPLETE;
+      return TaskStatus.READY_FOR_REVIEW;
```
Comment (garyjzhao) [inline src/views/ModelPlan/TaskList/index.tsx:120]: This is a temporary solution until https://github.com/CMSgov/mint-app/pull/162

File: src/views/ModelPlan/TaskList/Beneficiaries/Frequency/index.tsx [hunk 2]
```diff
@@ -48,7 +49,13 @@ const Frequency = () => {
   const { t: h } = useTranslation('draftModelPlan');
   const { modelID } = useParams<{ modelID: string }>();
 
-  const formikRef = useRef<FormikProps<FrequencyFormType>>(null);
+  // Omitting readyForReviewBy and readyForReviewDts from initialValues and getting submitted through Formik
+  type InitialValueType = Omit<
+    FrequencyFormType,
+    'readyForReviewBy' | 'readyForReviewDts'
+  >;
```
Comment (garyjzhao) [inline src/views/ModelPlan/TaskList/Beneficiaries/Frequency/index.tsx:56]: This is omitting `readyForReviewBy` and `readyForReviewDts` to be submitted in the formik and thus throwing an error with the update mutations. 

This is in each task list

**Review by patrickseguraoddball — CHANGES_REQUESTED** [review]
Nice work Gary, this works great.  If you could give your `ReadyForReview` component a unit test, then I can approve!

**Review by patrickseguraoddball — APPROVED** [review]
Thanks for the unit test Gary!

**Pull request merged**